### PR TITLE
update to 1.2.22

### PR DESCRIPTION
**Pull request by ms32035**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -73,5 +72,7 @@ RUN \
     openssl-devel mariadb-devel sendmail curl wget help2man perl-libwww-perl && \
     yum clean all && \
     rm -rf /var/cache/yum/* && \
+    sed -i '/session    required   pam_loginuid.so/c\#session    required   pam_loginuid.so' /etc/pam.d/crond && \
```
Comment (proxac) [inline Dockerfile:75]: I'm curious why you comment out this section.  Was there a specific problem you were trying to resolve?

File: Dockerfile [hunk 2]
```diff
@@ -73,5 +72,7 @@ RUN \
...
+    chmod 0644 /etc/crontab && \
```
Comment (proxac) [inline Dockerfile:76]: Nice catch.  The permissions on `/etc/crontab` were preventing the poller from running

**Review by proxac — COMMENTED** [review]
I've been quietly using this container for a while now.  Huge thanks to everyone who contributes!

I tested this PR with the update to 1.2.22.  I ran into all kinds of issues with recent changes in Cacti to `csrf`.  Looking at the changes in the Cacti release, it appears additional changes may be required in the Docker `start.sh` script to ensure that a `csrf-secret.php` file is correctly handled.

I also noticed the change in this PR to switch the poller to `root`.  I ran into this same issue testing [1100524](https://github.com/scline/docker-cacti/commit/1100524e83075553ad7c5bfbd22a42f6caab22db).  After a lot of research and weighing pros/cons, I ultimately decided the better approach was to make `/bin/ping` a set-uid binary.  Setting the bit not only prevents having to run the poller as root, but also allows the UI to use ICMP when configuring devices.

File: Dockerfile (hunk 1 above)
Comment (ms32035) [reply Dockerfile:75]: Before switching to root I had an issue running as apache too. https://stackoverflow.com/questions/21926465/issues-running-cron-in-docker-on-different-hosts Don't remember exact circumstances, so might not be needed anymore

**Comment by ms32035** [comment]
correct, with csrf secrets I just manually changed the permissions once. 

How do you want to proceed with this? Feel free to pick the changes and apply to another branch if you want to. Otherwise you can also push your stuff here.

**Comment by ms32035** [comment]
Ok, I incorporated all the changes and later tested the upgrade script to upgrade to 1.2.23. Everything looks to be working ok

File: Dockerfile (hunk 1 above)
Comment (proxac) [reply Dockerfile]: Thanks for the additional info.  I can confirm the poller ran for me after you removed this change.

**Comment by proxac** [comment]
I tested with your latest changes.  `cron` is again running the poller.  And the issues with csrf are resolved.

Now I'm seeing the poller timeout and throw a bunch of PHP errors:

```
2023-01-02 13:30:01 - CMDPHP PHP ERROR Backtrace: (CactiShutdownHandler())
2023-01-02 13:30:01 - ERROR PHP ERROR: Uncaught TypeError: Unsupported operand types: array & bool in /cacti/lib/mib_cache.php:344 Stack trace: #0 /cacti/lib/mib_cache.php(384): MibCache->delete() #1 /cacti/lib/snmpagent.php(352): MibCache->replace() #2 /cacti/poller.php(767): snmpagent_poller_bottom() #3 {main} thrown in file: /cacti/lib/mib_cache.php on line: 344
2023-01-02 13:30:01 - CMDPHP PHP ERROR WARNING Backtrace: (/poller.php[705]:snmpagent_poller_exiting(), /lib/snmpagent.php[302]:CactiErrorHandler())
2023-01-02 13:30:01 - ERROR PHP WARNING: Trying to access array offset on value of type bool in file: /cacti/lib/snmpagent.php on line: 302
```

Is Cacti ready for the version of PHP bundled with Rocky 9?

**Comment by ms32035** [comment]
Never seen this, looks more like something specific to your installation.

**Comment by scline** [comment]
Thanks for the PR, will do some basic checks this weekend and if all looks good will merge before Monday (^_^)

**Comment by ms32035** [comment]
@scline did you manage to test it?

**Pull request merged**